src/test/java/me/zeroeightsix/fiber/api/schema/type/derived/ConfigTypesTest.java

```diff
@@ -1,22 +1,84 @@
 package me.zeroeightsix.fiber.api.schema.type.derived;
 
-import static java.util.stream.Collectors.toList;
-import static org.junit.jupiter.api.Assertions.assertArrayEquals;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import me.zeroeightsix.fiber.api.builder.ConfigLeafBuilder;
+import me.zeroeightsix.fiber.api.exception.FiberConversionException;
+import me.zeroeightsix.fiber.api.schema.type.DecimalSerializableType;
+import me.zeroeightsix.fiber.api.schema.type.ListSerializableType;
+import me.zeroeightsix.fiber.api.tree.ConfigLeaf;
+import me.zeroeightsix.fiber.api.tree.PropertyMirror;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
 
 import java.math.BigDecimal;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
-import me.zeroeightsix.fiber.api.exception.FiberConversionException;
-import org.junit.jupiter.api.Test;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ConfigTypesTest {
+
+	@DisplayName("Test numerical constraints")
+	@Test
+	void testNumericalConstraints() {
+		assertThrows(NullPointerException.class, () -> ConfigTypes.UNBOUNDED_DECIMAL.withIncrement(5), "Invalid increment");
+		assertThrows(IllegalArgumentException.class, () -> ConfigTypes.INTEGER.withIncrement(-5), "Invalid increment");
+
+		DecimalSerializableType type = ConfigTypes.NATURAL.withMinimum(10).withMaximum(20).getSerializedType();
+
+		Predicate<Integer> finalConstraint = i -> type.accepts(BigDecimal.valueOf(i));
+
+		assertEquals(BigDecimal.TEN, type.getMinimum(), "Correct minimum");
+		assertEquals(BigDecimal.valueOf(20), type.getMaximum(), "Correct maximum");
+
+		assertFalse(finalConstraint.test(-2), "Input can't be lower than 10");
+		assertFalse(finalConstraint.test(4), "Input can't be lower than 10");
+
+		assertTrue(finalConstraint.test(15), "Input can be between 10 and 20");
+
+		assertFalse(finalConstraint.test(25), "Input can't be above 20");
+		assertFalse(type.accepts(BigDecimal.valueOf(0.5)));
+	}
+
+	@DisplayName("Test array type constraints")
+	@Test
+	void testArrayConstraints() {
+		DecimalSerializableType elementType = ConfigTypes.INTEGER.withValidRange(3, 10, 2).getSerializedType();
+		ListSerializableType<BigDecimal> type = new ListSerializableType<>(elementType, 0, 3, false);
+		Predicate<Integer[]> config = arr -> type.accepts(Arrays.stream(arr).map(BigDecimal::new).collect(Collectors.toList()));
+
```

Comment:
`BigDecimal::valueOf` and plain `toList()`

Reply:
plain `toList()` ?

Reply:
`toList()` is already statically imported.

Reply:
Well, since we are mostly avoiding static imports in this project I'll switch it to a regular import. We can always decide on a different convention later.